# Generated by Django 2.1.2 on 2018-11-16 09:31

from django.conf import settings
from django.db import migrations, models
import django.db.models.deletion


class Migration(migrations.Migration):

    dependencies = [
        ('account', '0030_auto_20181116_1344'),
    ]

    operations = [
        migrations.CreateModel(
            name='Follow',
            fields=[
                ('id', models.AutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
                ('follow_time', models.DateTimeField(auto_now=True)),
                ('follower', models.ForeignKey(on_delete=django.db.models.deletion.CASCADE, related_name='follower_user', to=settings.AUTH_USER_MODEL)),
                ('following', models.ForeignKey(on_delete=django.db.models.deletion.CASCADE, related_name='follow_user', to=settings.AUTH_USER_MODEL)),
            ],
        ),
        migrations.AlterUniqueTogether(
            name='relation',
            unique_together=set(),
        ),
        migrations.RemoveField(
            model_name='relation',
            name='from_user',
        ),
        migrations.RemoveField(
            model_name='relation',
            name='to_user',
        ),
        migrations.RemoveField(
            model_name='profile',
            name='follow_set',
        ),
        migrations.DeleteModel(
            name='Relation',
        ),
        migrations.AddField(
            model_name='user',
            name='follow_set',
            field=models.ManyToManyField(blank=True, related_name='followers', through='account.Follow', to=settings.AUTH_USER_MODEL),
        ),
        migrations.AlterUniqueTogether(
            name='follow',
            unique_together={('following', 'follower')},
        ),
    ]
